### DISCUSSION TITLE
有没有可能用 GitHub Actions 来代替我们完成重复的素材资源同步任务

### DISCUSSION THREAD

**Discussion by mark9804** [Arondight/Adachi-BOT#480] [Ideas]
资源同步脚本每个版本都要手动执行至少一次，有没有可能让 [Actions](https://github.com/Arondight/Adachi-BOT/actions/new) 来代替人工来完成这个工作？我感觉这个同步脚本执行还挺耗时的，不如交给 Actions 让它自己每天早上六七点的时候跑一遍，然后有改动就提交commit或者PR，没改动就终止，给我们这边省下的时间拿来夜宵加个鸡蛋

**Comment by Arondight** [comment]
这个应该是可以搞的，不过得找找有没有现成的 action ，要不然得手撸代码以用我的 key 去操作 repo 稍微麻烦一点

**Reply by Arondight** [reply]
而且好像没找到 GitHub 的 node 库？官方有个 python 库，但是直接把 python 放到一个 js 项目里会很奇怪，新建一个 repo 当 action 使用比较好

**Reply by Arondight** [reply]
有了，现成的 action https://github.com/actions/checkout

**Comment by Arondight** [comment]
现在 actions 都搞完了，唯一的问题就是这个 actions 运行时间很长（大约十分钟），然后每一个 fork 都会在每天同一个时间跑，希望 github 不要把这个项目直接给禁掉了，所有 fork 一起消失

**Reply by mark9804** [reply]
`npm run check` 和 `npm run format` 这两个 actions 不是问题，同步资源算是一个起定期维护功能的脚本，应该也和 serverless 扯不上关系，现在的问题是同步资源算不算 “和 production, testing, deployment or publication” 无关的 actions，我感觉还是相关的，虽然每天跑一次确实频繁了点

> Actions should not be used for:
> * any activity that places a burden on our servers, where that burden is disproportionate to the benefits provided to users (for example, don't use Actions as a content delivery network or as part of a serverless application, but a low benefit Action could be ok if it’s also low burden); or
> * if using GitHub-hosted runners, any other activity unrelated to the production, testing, deployment, or publication of the software project associated with the repository where GitHub Actions are used.

**Comment by Arondight** [comment]
改了 tunmb 的同步变为不更新已有文件，现在四分钟多一点了

https://github.com/Arondight/Adachi-BOT/runs/4511485902?check_suite_focus=true
